### PR TITLE
Fix updated issue-regex

### LINKED ISSUE #12: `issue-regex@4.1.0` causes dom to fail
_**Continued discussion from https://github.com/refined-github/refined-github/issues/6930**_

https://github.com/sindresorhus/issue-regex/commit/92a86c81dbfde3e44eade86a732b66d718f189b5 changed the capturing group regex causing this logic ↓ to stop working properly

https://github.com/sindresorhus/linkify-issues/blob/8cfc515a0007d0e297e2a37bfd2bfa8d584b6ae4/index.js#L31-L37

### PR DESCRIPTION
**Pull request by Katsute**
Fixes #12 

* Issues are always at `groupCount n + 1`
* Text is always non-matching at `groupCount n + 0`

`See #1 for more info`

```
5th pos →   [0] 'See '
2nd pos →   [1] '#1'
            [2] 'undefined'
            [3] 'undefined'
            [4] '1'
5th pos →   [5] ' for more info'
```

`#1 fixes refined-github/refined-github#6930`

```
5th pos →   [0] ''
2nd pos →   [1] '#1'
            [2] 'undefined'
            [3] 'undefined'
            [4] '1'
5th pos →   [5] ' fixes '
2nd pos →   [6] 'refined-github/refined-github#6930'
            [7] 'refined-github'
            [8] 'refined-github'
            [9] '6930'
5th pos →   [10] ''
```

### CODE REVIEW

**Review by sindresorhus — APPROVED** [review]

File: index.js [hunk 1]
```diff
@@ -28,10 +28,12 @@ const getAsString = (string, options) => string.replace(groupedIssueRegex, match
 
 const getAsDocumentFragment = (string, options) => {
 	const fragment = document.createDocumentFragment();
-	for (const [index, text] of Object.entries(string.split(groupedIssueRegex))) {
-		if (index % 2) { // URLs are always in odd positions
+	const parts = string.split(groupedIssueRegex);
+
+	for (const [index, text] of parts.entries()) {
+		if (index % 5 === 1) { // Issues are always first in capturing group
```
Comment (fregante) [inline index.js]: MAGIC_NUMBER_ALERT

`issue-regex` must be use a fixed version in order to guarantee this. Next thing you know someone alters the group count in the package and this fails again.

There should also be a `const issueRegexGroups = 5;` constant

File: index.js [hunk 2]
```diff
@@ -28,10 +28,12 @@ const getAsString = (string, options) => string.replace(groupedIssueRegex, match
...
 			fragment.append(domify(linkify(text, options)));
-		} else if (text.length > 0) {
+		} else if (index % 5 === 0 && text.length > 0) { // Text is always in split position
```
Comment (fregante) [inline index.js]: Neither comment here explains the situation. It should be more like:

> Every regex group appears in the split strings
> At position 5n+1 there's the full matched issue string
> At position 5n+0 there's what doesn't match the regex

File: package.json [hunk 3]
```diff
@@ -38,7 +38,7 @@
 	],
 	"dependencies": {
 		"create-html-element": "^4.0.1",
-		"issue-regex": "^4.0.0"
+		"issue-regex": "^4.1.0"
```
Comment (fregante) [inline package.json:41]:
```suggestion
		"issue-regex": "4.1.0"
```

**Review by fregante — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (fregante) [reply index.js]: > Next thing you know

…a PR for this is submitted, 4.2.0 is published, this loop breaks again.

- https://github.com/sindresorhus/issue-regex/issues/15

File: index.js (hunk 2 above)
Comment (Katsute) [reply index.js]: [`8325460` (#13)](https://github.com/sindresorhus/linkify-issues/pull/13/commits/8325460e646bc3b94cc093e955dd01c8a4552fd8)

File: index.js (hunk 1 above)
Comment (Katsute) [reply index.js]: [`8325460` (#13) (L5)](https://github.com/sindresorhus/linkify-issues/pull/13/commits/8325460e646bc3b94cc093e955dd01c8a4552fd8#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346R5) returns the group count

Comment (fregante) [reply index.js]: Smart! 😂 no need to pin the version then

Comment (Katsute) [reply index.js]: I would still actually pin the version, in case the order of groups changes

Comment (fregante) [reply index.js]: I don’t think that changes since the first group is what we create on line 4, there can't be an earlier group

Comment (Katsute) [reply index.js]: Reverted in [`36df55d` (#13)](https://github.com/sindresorhus/linkify-issues/pull/13/commits/36df55d88146d2bc791b9165d4b6084e682bce8f)

**Review by fregante — APPROVED** [review]

File: index.js [hunk 4]
```diff
@@ -2,6 +2,7 @@ import issueRegex from 'issue-regex';
 import createHtmlElement from 'create-html-element';
 
 const groupedIssueRegex = new RegExp(`(${issueRegex().source})`, 'g');
+const issueRegexGroups = (new RegExp(groupedIssueRegex.toString() + '|')).exec('').length; // Number of capturing groups in regex
```
Comment (fregante) [inline index.js:5]: Just saw: why toString and not source?

Comment (fregante) [reply index.js:5]: Also no need to wrap the constructor. `new RegExp().exec()` is perfectly fine

Comment (Katsute) [reply index.js:5]: I was following https://stackoverflow.com/a/16046903 while trying to figure out how to retrieve the group count. The only different seems to be that `toString` includes the `/.../g` flag, which doesn't affect the capturing groups but is unnecessary.

Fix opened as https://github.com/sindresorhus/linkify-issues/pull/14

**Pull request merged**